src/eei.cpp

```diff
@@ -275,7 +275,7 @@ namespace HeraVM {
       evm_tx_context tx_context;
       context->fn_table->get_tx_context(&tx_context, context);
 
-      return Literal((int64_t)tx_context.block_gas_limit);
+      return Literal((uint64_t)tx_context.block_gas_limit);
```

Comment:
Use `static_cast`.

Reply:
Actually these can stay `int64_t`, but is there a way to just throw an error if they are not `int64_t` without typecasting?